components/apimgt/org.wso2.carbon.apimgt.cleanup.service/src/main/java/org/wso2/carbon/apimgt/cleanup/service/organizationPurge/ApplicationPurge.java

```diff
@@ -17,7 +17,98 @@
  */
 package org.wso2.carbon.apimgt.cleanup.service.organizationPurge;
 
-public class ApplicationPurge implements OrganizationPurge {
-    public void deleteOrganization(String organization) {
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.apimgt.api.APIManagementException;
+import org.wso2.carbon.apimgt.api.model.Subscriber;
+import org.wso2.carbon.apimgt.impl.APIConstants;
+import org.wso2.carbon.apimgt.impl.dao.ApiMgtDAO;
+
+import javax.cache.Cache;
+import javax.cache.Caching;
+import java.util.*;
+
+public class ApplicationPurge implements OrganizationPurge{
+    protected ApiMgtDAO apiMgtDAO;
+    private static final Log log = LogFactory.getLog(ApplicationPurge.class);
+
+    public ApplicationPurge() {
+        apiMgtDAO = ApiMgtDAO.getInstance();
+    }
+
+    public ApplicationPurge(ApiMgtDAO apiMgtDAO) {
+        this.apiMgtDAO = apiMgtDAO;
+    }
+    /**
+     * Delete organization related application data
+     *
+     * @param organization organization
+     * @throws APIManagementException if failed to clean up organization
+     */
+    @Override public void deleteOrganization(String organization) throws APIManagementException {
```

Comment:
We are not actually deleting the organization, but deleting application specific data of a particular organization
```suggestion
    @Override 
    public void purge(String organization) throws APIManagementException {
```